framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,15 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
> isn't that the job of Yii::$app->charset?

Yes it is - this is a job, which makes `Application` instance to be essential for the helper running.

>Ternary check only makes the default value available if app is not configured.

You are introducig magick constant with it, which should never be present in the code.
Standalone valdator and helper running will become impossible in case you need default encoding different from `UTF-8`.